chardevice.go

```diff
@@ -170,10 +170,16 @@ func (cdev CharDevice) QemuParams(config *Config) []string {
 	}
 
 	// Legacy serial is special. It does not follow the device + driver model
-	if cdev.Driver != LegacySerial {
+	if cdev.Driver != LegacySerial && cdev.Driver != PCISerial {
```

Comment:
Yeah, I don't like extending this, but will keep it for now.

In serialdevice_test.go, the VirtioSerialPortString test uses a CharDev object to emit the -device virtseiralport, but really, we should construct two objects:

```
chardev := CharDevice{
    // Driver:   VirtioSerialPort, // omit this here
    Backend:  Socket,
    ID:       "char0",
    Path:     "/tmp/char.sock",
}

serial := SerialDevice{
    Driver:    VirtioSerialPort,
    ID:        "channel0"
    CharDevID: "char0",
    Name:      "channel.0",
}

c := &Config{
    CharDevices:   []CharDevice{chardev},
    SerialDevices: []SerialDevice{serial},
   }
```